core/config/uniAssayConfig.js

```diff
@@ -4,15 +4,29 @@ import { noCustomization } from './noCustomization';
 import { makeCoreMintKeeper } from './coreMintKeeper';
 import { makeUniStrategy } from './strategies/uniStrategy';
 
+import { insist } from '../../util/insist';
+import { mustBeComparable } from '../../util/sameStructure';
+
 // This UniAssay config is used to create invites and similar assets.
 // It does not customize the purses, payments, mints, or issuers, and
 // it uses the core mintKeeper as well as the uniAssay (naturally)
-function makeUniAssayConfigMaker(descriptionCoercer) {
+const makeCustomInsistKind = descriptionCoercer => optDescription => {
```

Comment:
This syntax 

```
const makeCustomFn = foo => bar => {
   ... bar ... foo ...
}

function n(foo) {
  const customFn = makeCustomFn(foo);
  ... customFn ...
}
```

looks convoluted to me, though I was able to understand it.  Would it better as

```
function n(foo) {
  function customFn(bar) {
   ... bar ... foo ...
};
  ... customFn ...
}
```

I guess this comes down to two style questions. 

1) How recognizable is ```foo => bar => { ... }``` as syntax for currying to javascript developers? It seemed weird but convenient to me, but I don't know what's common.

2) Is it helpful to write a currying function even though the value to be curried is immediately available? In this case, it does have the advantage of moving 8 lines of code outside a function definition.

Reply:
Hmm, I don't know if I can answer how recognizable it is. A [JavaScript book](https://leanpub.com/javascriptallongesix/read) that I really like (and has been on the front page of Hacker News a couple of times) uses it a lot. The functional programming community uses it a lot. I think it adds more clarity than having `makers` for `makers`, but that could be just me.

Re: 2: I could have put that whole function inside `makeUniAssayConfigMaker` as you say, relying on `descriptionCoercer` to be in scope. I tend not to like this kind of entanglement and reliance on scope because it makes it harder to test and refactor, IMO. What do you think?

Reply:
re: #1. Thanks for the reference. I'm happy to use this as justification for what JS programmers should be familiar with. I'll try to find time to read more of it so I can have an opinion on whether we should recommend it as a style guide in general.

re: #2. I was about to react to the "harder to test" part of your response, and say that this would be a strong argument when used to actually write tests, but then I noticed "and refactor". I now agree that this is clearer and easier to manipulate, even though the variable is referenced both abstractly and concretely. Having the definition and use so close to one another also arguably makes it easier to see what's going on.